File: plant_nursery_management_system/src/main/java/com/project/plant_nursery_management_system/service/serviceInterface/CartService.java
package com.project.plant_nursery_management_system.service.serviceInterface;

import com.project.plant_nursery_management_system.dto.CartItemDto;
import com.project.plant_nursery_management_system.model.Cart;

public interface CartService {
    Cart addToCart(Long userId, CartItemDto itemDto);
    Cart updateQuantity(Long userId, CartItemDto itemDto);
    Cart removeFromCart(Long userId, Long plantId);
    void clearCart(Long userId);
    Cart getCartByUserId(Long userId);
    void deleteCart(Long userId);
}

File: PNMS/plant-nusery/src/services/cartService.js
import axiosInstance from "./axiosInstance";

const BASE_URL = "/cart";

export const cartService = {
  getCart: (userId) => axiosInstance.get(`${BASE_URL}/${userId}`),

  addToCart: (userId, cartItem) =>
    axiosInstance.post(`${BASE_URL}/${userId}/add`, cartItem),

  updateQuantity: (userId, cartItem) =>
    axiosInstance.put(`${BASE_URL}/${userId}/update`, cartItem),

  removeFromCart: (userId, plantId) =>
    axiosInstance.delete(`${BASE_URL}/${userId}/remove/${plantId}`),

  clearCart: (userId) => axiosInstance.delete(`${BASE_URL}/${userId}/clear`),

  deleteCart: (userId) => axiosInstance.delete(`${BASE_URL}/${userId}`),
};
